client/serviceinterface_remove.go

```diff
@@ -19,11 +21,14 @@ func (cli *VanClient) ServiceInterfaceRemove(ctx context.Context, address string
 		} else {
 			delete(current.Data, address)
 			_, err = cli.KubeClient.CoreV1().ConfigMaps(cli.Namespace).Update(current)
+
 			if err != nil {
 				return fmt.Errorf("Failed to update skupper-services config map: %v", err.Error())
-			} else {
-				return nil
 			}
+
+			removeServiceCertificate(cli.Namespace, address, cli.KubeClient)
```

Comment:
Probably want to make this conditional on the tlsCredentials being defined for the service?

Reply:
You are correct! done